### ISSUE TITLE
Is it possible to authorize an address to trade on my behalf?

### ISSUE THREAD

**Issue by MeoMix** [etherdelta/etherdelta.github.io#271]
Heya,

So, chuggin' away still on this code. Making good progress. Thought I would give trading with my own contract a whirl.

I got the basics of it going and proxied a `trade` call through my own contract. I know the order was properly signed, but the trade failed. Presumably this is because `msg.sender` changes to my contract's address, rather than my address, which doesn't have funds.

I'm gearing up to make my contract smarter and be able to send/receive value + deposit/withdraw value into EtherDelta's contract -- effectively making it the trader.

I wanted to confirm this was necessary and that it's not possible to whitelist my contract's address for trading as my address. I note that there are `approve` methods on ED's token smart contract, but I don't think those are quite what I want, right?

**Comment by DeltaBalances** [comment]
Seems like you are finally getting to the fun parts, sending your funds to your contract, and then hoping that you can get them out again.

The only way to trade with the funds in your walet is by signing the trades with that private key directly. If you want to use a contract, the funds need to be deposited by the contract.

Just watch some of the trade histories, there are plenty of contracts running to give some inspiration. Its all private though, as nobody wants to really help the competition 😄.

I actually made my website just to better keep track of mine.

**Comment by MeoMix** [comment]
@DeltaBalances Oh that's funny. A friend was just showing me your website yesterday.

Yeah, I figure it's all private. No big there. I spent a couple of days doing a practice security audit of Ethfinex's Nectar token to get up to speed a bit. I've got my contract written at this point. About to test moving money into/out of it on the test net and trying to decide if I want to support moving alt coins in/out or if I think I can get it to always trade back to ETH before moving. :p
